### PR TITLE
[PATCH v9] Rework inlining headers 

### PR DESCRIPTION
**Pull request by lumag**
Rework platform-specific inline headers to remove necessity to call inlineable functions using underscore from platform code. This is just an RFC, changing two header files. The rest will come later, if overall approach is approved.

### CODE REVIEW

**Comment by lumag** [comment]
@muvarov @psavol I have updated this PR according to comments during arch call. Could you please review and provide feedback on this idea?

File: m4/odp_dpdk.m4 [hunk 1]
```diff
@@ -12,6 +12,12 @@ AS_CASE([$cur_driver],
     [rte_pmd_mlx4], [AS_VAR_APPEND([DPDK_LIBS], [" -lmlx4 -libverbs"])],
     [rte_pmd_mlx5], [AS_VAR_APPEND([DPDK_LIBS], [" -lmlx5 -libverbs"])],
     [rte_pmd_pcap], [AS_VAR_APPEND([DPDK_LIBS], [" -lpcap"])],
+    [rte_pmd_aesni_gcm], [AS_VAR_APPEND([DPDK_LIBS], [" -lIPSec_MB"])],
+    [rte_pmd_aesni_mb], [AS_VAR_APPEND([DPDK_LIBS], [" -lIPSec_MB"])],
+    [rte_pmd_kasumi], [AS_VAR_APPEND([DPDK_LIBS], [" -lsso_kasumi"])],
+    [rte_pmd_snow3g], [AS_VAR_APPEND([DPDK_LIBS], [" -lsso_snow3g"])],
+    [rte_pmd_zuc], [AS_VAR_APPEND([DPDK_LIBS], [" -lsso_zuc"])],
+    [rte_pmd_qat], [AS_VAR_APPEND([DPDK_LIBS], [" -lcrypto"])],
```
Comment (psavol) [inline m4/odp_dpdk.m4]: This patch should not be in this patch set.

File: platform/linux-generic/include/odp/api/plat/odp_inline.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * under different conditions */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @cond _ODP_HIDE_FROM_DOXYGEN_ */
+
+#undef _ODP_INLINE
+#ifdef _ODP_API_IMPL
```
Comment (psavol) [inline platform/linux-generic/include/odp/api/plat/odp_inline.h]: For example, _ODP_API_NOT_INLINED would be more understandable define name. This define is defined when xxx_inlines.h is used for not inlined implementation of the functions.

File: platform/linux-generic/Makefile.am [hunk 3]
```diff
@@ -26,13 +26,11 @@ odpapiplatinclude_HEADERS = \
...
 		  include/odp/api/plat/byteorder_inlines_api.h \
-		  include/odp/api/plat/packet_flag_inlines.h \
-		  include/odp/api/plat/packet_flag_inlines_api.h \
+		  include/odp/api/plat/packet_flags_inlines.h \
 		  include/odp/api/plat/packet_inline_types.h \
 		  include/odp/api/plat/packet_inlines.h \
 		  include/odp/api/plat/packet_inlines_api.h \
-		  include/odp/api/plat/pktio_inlines.h \
-		  include/odp/api/plat/pktio_inlines_api.h \
+		  include/odp/api/plat/packet_io_inlines.h \
```
Comment (psavol) [inline platform/linux-generic/Makefile.am]: The patch set would be easier to follow if file names would not be changed, or at least file name changes would be done in a separate patch. I would not change pktio file name as API uses odp_pktio_ prefix instead of packet_io, so packet_io should be changed to pktio - not the other way around.

**Comment by psavol** [comment]
I'm trying out similar macro based inlining scheme with odp_event_type(). It seems to work OK already, but exposed a linking issue in IPsec validation test. It's a bit simpler than this. I'll send it out next week. Most of the work here can be reused with that scheme also.

**Comment by lumag** [comment]
@psavol I thought about macro-based inlining, but decided to use function-based one. Let's see your patches though.

File: platform/linux-generic/Makefile.am (hunk 3 above)
Comment (lumag) [reply platform/linux-generic/Makefile.am]: Nice idea, thanks.

**Comment by Bill-Fischofer-Linaro** [comment]
@lumag Should this PR be withdrawn now that PR #593 has been merged?

**Comment by lumag** [comment]
@Bill-Fischofer-Linaro no, I will just rebase this on top of merged one.

**Comment by lumag** [comment]
@psavol @Bill-Fischofer-Linaro I've updated PR to reflect merged changes.

**Comment by lumag** [comment]
@psavol @muvarov is it ready to be merged?

**Comment by Bill-Fischofer-Linaro** [comment]
@lumag We agreed at today's PERF call that it is. I expect it to be merged soon.

**Comment by muvarov** [comment]
Merged.
